### ISSUE TITLE
`create-solid` does not support monorepo with yarn workspaces and hoisted deps

### ISSUE THREAD

**Issue by jorroll** [solidui/solid-start#4]
Using `create-solid` to setup a new project within a monorepo that uses [yarn workspaces](https://classic.yarnpkg.com/en/docs/workspaces/) and hoisted dependencies (which is the default) does not appear to work.

See reproduction here:
- https://gitlab.com/john.carroll.p/solidstart-monorepo-issue

Running `yarn install` and then `yarn dev` results in the following warning and the app does not serve

```
You appear to have multiple instances of Solid. This can lead to unexpected behavior.
computations created outside a `createRoot` or `render` will never be disposed
```

Usage within a monorepo without hoisted dependencies seems to work fine. However, I would like for all projects within the repo to use the same solidjs dependency.

### npm Workspaces

Testing with [npm v7 workspaces](https://docs.npmjs.com/cli/v7/using-npm/workspaces) also seems to have the same error, though I'm less familiar with npm workspaces so it's unclear if they are expected to work similarly to yarn workspaces.

**Comment by jorroll** [comment]
Upon further inspection, the issue appears to be that vite is [caching build artifacts](https://vitejs.dev/guide/dep-pre-bundling.html#customizing-the-behavior) in a special `node_modules/.vite` folder. In this case, build artifacts include `solid-js/dev`, `solid-js/web`, and `chunk-A...` modules. The `chunk-A...` module includes its own

```js
var DEV;
{
  DEV = {
    writeSignal,
    serializeGraph
  };
}
if (globalThis) {
  if (!globalThis.Solid$$)
    globalThis.Solid$$ = true;
  else
    console.warn("You appear to have multiple instances of Solid. This can lead to unexpected behavior.");
}
```

This is currently my best guess to explain how a second copy of `solid-js` is being created when only one version is installed.  Unfortunately, following vite's instructions to disable build dependency caching (`optimizeDeps.exclude`) and test if that fixes the problem does not seem to work.

**Comment by jorroll** [comment]
> That requirement cannot change and is true of most frameworks so have no idea why these workspace technologies seem to be defaulting otherwise.

It should be noted that the workspace only installs a single version of `solid-js`, so something else appears to be going wrong. In this case, it looks like the current `vite` build setup is the culprit (see above).

**Comment by jorroll** [comment]
Holy crap I figured it out!!!!!!!!!!!!!!!!!!!!!!!! 🎉 🎊 🥳 🎉 🥳

Or rather, I have a solution. To try to figure out what was going wrong, I compared the results of running `solid-start dev` in a regular repo vs the monorepo. I noticed that, in the regular repo, three dependencies were being pre-bundled `solid-js/web`, `solid-js`, and `solid-js/dev` but, in the monorepo, only two dependencies were being pre-bundled `solid-js/web`, `solid-js/dev`.

Adding `solid-js` to the vite config's `optimizeDeps.include` array got the monorepo to pre-bundle `solid-js/web`, `solid-js`, and `solid-js/dev` just like in the regular repo and it eliminated the warning that there were multiple instances of solid-js. It also finally got the default start page to load in a browser. Unfortunately, it was clear something was going wrong as reactivity on the webpage wasn't working and the page styles hadn't loaded. A console error indicated that

```html
<script
  type="module"
  src="/node_modules/solid-start/runtime/entry.jsx"
></script>
```

in `index.html` was failing to load. I realized this made sense, because the dev server root was the project root rather than the workspace root (and `solid-start` was installed in the workspace root). I tried changing the module link to

```html
<script
  type="module"
  src="../../node_modules/solid-start/runtime/entry.jsx"
></script>
```

But I discovered that the dev server couldn't load anything outside its root. However, by symlinking the `solid-start` package in the workspace root's `node_modules` into the project root's `node_modules`, then the dev server was able to load the code without issue and everything started working.

I'll put together a PR in a day or two.
